jerry-core/vm/vm.c

```diff
@@ -354,10 +328,30 @@ vm_construct_literal_object (vm_frame_ctx_t *frame_ctx_p, /**< frame context */
     }
 
     return ret_value;
-#else /* CONFIG_DISABLE_REGEXP_BUILTIN */
-    JERRY_UNREACHABLE (); /* Regular Expressions are not supported in the selected profile! */
+  }
 #endif /* !CONFIG_DISABLE_REGEXP_BUILTIN */
+
+  JERRY_ASSERT (bytecode_p->status_flags & CBC_CODE_FLAGS_FUNCTION);
+
+  ecma_object_t *func_obj_p;
+
+#ifndef CONFIG_DISABLE_ES2015_ARROW_FUNCTION
+  if (!(bytecode_p->status_flags & CBC_CODE_FLAGS_ARROW_FUNCTION))
+  {
+#endif /* !CONFIG_DISABLE_ES2015_ARROW_FUNCTION */
```

Comment:
Can't re reorganize this with an `unlikely` to use the ifdef only once?

Reply:
An `unlikely` is a runtime condition even if the branch predictor gets hints from the developer. An `#if[n]def` guarding out that condition surely wont execute a code that's not even there.